src/categorical_algebra/CSets.jl

```diff
@@ -550,10 +553,12 @@ struct BacktrackingState{S <: SchemaDescType,
   dom::Dom
   """ Codomain ACSet: the "values" in the CSP. """
   codom::Codom
+  loose::LooseFun
 end
 
 function backtracking_search(f, X::StructACSet{S}, Y::StructACSet{S};
-                             monic=false, iso=false, initial=(;)) where {Ob, S<:SchemaDescType{Ob}}
+                             monic=false, iso=false, loose=(;), initial=(;),
```

Comment:
I think we should rename the kw arg `loose` to `type_components`, since that's used elsewhere in this file.